### PR TITLE
matrix-appservice-slack: 1.11.0 -> 2.0.1

### PR DESCRIPTION
**Pull request by chvp**
###### Description of changes

Includes https://github.com/NixOS/nixpkgs/pull/194365
Supercedes #194622

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chvp** [comment]
Waiting on https://github.com/NixOS/nixpkgs/pull/191129, since that makes packaging this a lot easier.

File: pkgs/servers/matrix-synapse/matrix-appservice-slack/default.nix [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
     owner = "matrix-org";
     repo = "matrix-appservice-slack";
-    rev = "1.11.0";
-    sha256 = "U1EHL1ZwcpCXA9sjya6ry/3Q+gwdQWPUDFN+wp1qjrg=";
+    rev = version;
+    hash = "sha256-iUoZO39v5jvskpF3Rrc0sfeNAbRlqiVa6wFOO8yWH5o=";
   };
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
```
Comment (winterqt) [inline pkgs/servers/matrix-synapse/matrix-appservice-slack/default.nix:21]: Can this use `fetchYarnDeps` instead? See [here](https://github.com/NixOS/nixpkgs/blob/27a89ba43b0fb735ce867e8ab3d2442f8cc61dad/pkgs/servers/monitoring/matrix-alertmanager/default.nix) for an example, and [here](https://github.com/NixOS/nixpkgs/blob/27a89ba43b0fb735ce867e8ab3d2442f8cc61dad/pkgs/applications/networking/instant-messengers/element/update.sh) for how to get the hash in an update script.

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/servers/matrix-synapse/matrix-appservice-slack/default.nix (hunk 1 above)
Comment (chvp) [reply pkgs/servers/matrix-synapse/matrix-appservice-slack/default.nix]: done

**Comment by chvp** [comment]
That's annoying, since I'm setting the pname and version myself anyway... but anyhow, fixed.

**Comment by winterqt** [comment]
Please rebase, as the matrix-sdk-crypto-nodejs PR has been merged.

**Comment by chvp** [comment]
Deployed the rebase to my server, still works as expected.

**Review by winterqt — APPROVED** [review]

**Pull request merged**